integration/tests/windows_managed/windows_managed_test.go

```diff
@@ -0,0 +1,199 @@
+	}
+
+	deleteCluster := func(clusterName string) {
+		By("deleting the windows cluster")
+		cmd := params.EksctlDeleteCmd.WithArgs(
+			"cluster",
+			"--name", clusterName,
+		)
+		Expect(cmd).To(RunSuccessfully())
+	}
+
+	Context("Create Windows managed nodegroup with taints", func() {
+		It("should create Windows managed nodegroup taints applied", func() {
+			taints := []api.NodeGroupTaint{
+				{
+					Key:    "key1",
+					Value:  "value1",
+					Effect: "NoSchedule",
+				},
+				{
+					Key:    "key2",
+					Effect: "NoSchedule",
+				},
+				{
+					Key:    "key3",
+					Value:  "value2",
+					Effect: "NoExecute",
+				},
+			}
+			clusterConfig := makeClusterConfig(params.ClusterName)
+			clusterConfig.ManagedNodeGroups = []*api.ManagedNodeGroup{
+				{
+					NodeGroupBase: &api.NodeGroupBase{
+						Name:         "windows-taints",
+						AMIFamily:    api.NodeImageFamilyWindowsServer2022CoreContainer,
+						VolumeSize:   aws.Int(120),
+						InstanceType: "t3a.xlarge",
+					},
+					Taints: taints,
+				},
+			}
+
+			cmd := params.EksctlCreateCmd.
+				WithArgs(
+					"nodegroup",
+					"--verbose", "4",
+					"--config-file", "-",
+				).
+				WithoutArg("--region", params.Region).
+				WithStdin(clusterutils.Reader(clusterConfig))
+			Expect(cmd).To(RunSuccessfully())
+
+			config, err := clientcmd.BuildConfigFromFlags("", params.KubeconfigPath)
+			Expect(err).NotTo(HaveOccurred())
+			clientset, err := kubernetes.NewForConfig(config)
+			Expect(err).NotTo(HaveOccurred())
+
+			mapTaints := func(taints []api.NodeGroupTaint) []corev1.Taint {
+				var ret []corev1.Taint
+				for _, t := range taints {
+					ret = append(ret, corev1.Taint{
+						Key:    t.Key,
+						Value:  t.Value,
+						Effect: t.Effect,
+					})
+				}
+				return ret
+			}
+			tests.AssertNodeTaints(tests.ListNodes(clientset, "taints"), mapTaints(taints))
+		})
+	})
+	Context("Create Windows managed nodegroup via CLI", func() {
+		It("should not return error when creating Windows managed nodegroup via CLI", func() {
+			cmd := params.EksctlCreateCmd.WithArgs(
+				"nodegroup",
+				"--cluster", params.ClusterName,
+				"--nodes", "4",
+				"--managed",
+				"--instance-types", "t3a.xlarge",
+				"--node-ami-family=WindowsServer2019CoreContainer",
+			)
+			Expect(cmd).To(RunSuccessfully())
+		})
+	})
+	Context("Create Windows managed nodegroups and run pods then cleanup", func() {
+		DescribeTable("it should be able to run Windows pods", func(windowsAmiType, workload, clusterName string) {
+			createClusterWithWindowsMNG(clusterName, windowsAmiType)
+			runWindowsPod(workload)
+			deleteCluster(clusterName)
+		},
+			Entry("Windows Server 2019 Full", api.NodeImageFamilyWindowsServer2019CoreContainer, "windows-server-iis.yaml", "managed-windows-cluster1"),
+			Entry("Windows Server 2022 Full", api.NodeImageFamilyWindowsServer2022FullContainer, "windows-server-iis-2022.yaml", "managed-windows-cluster2"),
+		)
+	})
```

Comment:
I think we don't need to have a test for creating two more clusters with different AMI families. We are wary of creating additional clusters in our integration tests as it slows them down. We can add the `runWindowsPod` call to the tests above after the nodegroup is created, and remove this entire `Context` block. What do you think?

Reply:
That makes sense. I will make another commit to remove the entire block. Also the lint check failed a few times, I was wondering if you have any suggestion on how to fix that?

Reply:
Can you modify the `create cluster` call in `BeforeSuite` to include Windows nodegroups by using a config file? That'll help us test `runWindowsPod`.